docs/recipes/typescript.md

```diff
@@ -8,13 +8,74 @@ This guide assumes you've already set up TypeScript for your project. Note that
 
 ## Enabling AVA's support for TypeScript test files
 
+### With precompile step
+
 Out of the box AVA does not load TypeScript test files. You can use our [`@ava/typescript`] package, which is designed to work for projects that precompile TypeScript using the `tsc` command. Please see [`@ava/typescript`] for setup instructions.
 
 ### Using `ts-node`
 
-You can use [`ts-node`] to do live testing without transpiling to js files.  This can be especially helpful when you're using a bundler.
+You can use [`ts-node`] to do live testing without transpiling. This can be especially helpful when you're using a bundler. Be sure to install the required dev dependencies:
+
+`npm install --save-dev typescript ts-node`
+
+Then, depending on whether or not your package is of type `module` or not, the required setup differs. See either:
+
+1. [for packages with type "module"](#for-packages-with-type-module)
+2. [for packages without type "module"](#for-packages-without-type-module)
+
+#### for packages with type `module`
+
+If your `package.json` has `"type": "module"`, then this is the AVA configuration you need:
+
+`package.json`:
+
+```json
+{
+	"ava": {
+		"extensions": {
+			"ts": "module"
+		},
+		"nonSemVerExperiments": {
+			"configurableModuleFormat": true
+		},
+		"nodeArguments": [
+			"--loader",
+			"ts-node/esm"
```

Comment:
Would this work?

```suggestion
			"--loader=ts-node/esm"
```

Reply:
I can try!

Reply:
@novemberborn yes this worked
<img width="893" alt="image" src="https://user-images.githubusercontent.com/3253920/110263137-1290dd00-7ff9-11eb-8b3b-e0d9a29003b9.png">